tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#define STACK_SIZE 512
+#define NUM_THREAD 3
+static char __noinit __stack tstack[NUM_THREAD][STACK_SIZE];
+/* slice size in millisecond*/
+#define SLICE_SIZE 200
+/* busy for more than one slice*/
+#define BUSY_MS (SLICE_SIZE+20)
+/* a half timeslice*/
+#define HALF_SLICE_SIZE (SLICE_SIZE >> 1)
+
+K_SEM_DEFINE(sema, 0, NUM_THREAD);
+/*elapsed_slice taken by last thread*/
+static s64_t elapsed_slice;
+/*expected elapsed duration*/
+static s64_t expected_slice[NUM_THREAD] = {
+	HALF_SLICE_SIZE,/* the ztest native thread taking a half timeslice*/
+	SLICE_SIZE,     /* the spawned thread taking a full timeslice, reset*/
+	SLICE_SIZE      /* the spawned thread taking a full timeslice, reset*/
+};
+static int thread_idx;
+
+static void thread_tslice(void *p1, void *p2, void *p3)
+{
+	s64_t t = k_uptime_delta(&elapsed_slice);
+
+	TC_PRINT("thread[%d] elapsed slice %lld, ", thread_idx, t);
+	TC_PRINT("expected %lld\n", expected_slice[thread_idx]);
```

Comment:
print things to the screen only in debug mode, the test output should  be clean without noise coming out of the test